File: test/go/gomodule/http-service/success-json-api/internal/handler/router.go
package handler
import (
    "encoding/json"
    "net/http"
    "github.com/go-chi/chi/v5"
    "example.com/windowstolinux/fixture/internal/config"
    "example.com/windowstolinux/fixture/internal/service"
)
func New(configuration config.Configuration) http.Handler {
    router := chi.NewRouter()
    serve := func(response http.ResponseWriter, request *http.Request) {
        status, body, contentType := configuration.Status, []byte(configuration.Label), "text/plain; charset=utf-8"
        if status != 503 && (request.URL.Path == "/api/summary" || configuration.Mode == "json") {
            var raw *string
            if request.URL.Path == "/api/summary" {
                if values, found := request.URL.Query()["values"]; found { raw = &values[0] }
            }
            summary, err := service.Summarize(raw)
            if err != nil { status, body = 400, []byte("invalid-values") } else {
                body, err = json.Marshal(summary)
                if err != nil { http.Error(response, "serialization-failed", 500); return }
                contentType = "application/json; charset=utf-8"
            }
        }
        response.Header().Set("Content-Type", contentType)
        response.WriteHeader(status)
        _, _ = response.Write(body)
    }
    router.Get("/api/summary", serve)
    router.Get("/", serve)
    router.Get("/*", serve)
    return router
}

File: test/single-language/ruby/rubycli/http-service/success-runtime-config/lib/router.rb
require 'uri'
require_relative 'summary_service'
module Fixture
  class Router
    def initialize(configuration)
      @configuration = configuration
    end

    def route(path, query)
      status, body, type = @configuration.status, @configuration.label, 'text/plain; charset=utf-8'
      if status != 503 && (path == '/api/summary' || @configuration.mode == 'json')
        begin
          pair = URI.decode_www_form(query || '').find { |key, _| key == 'values' }
          body = SummaryService.new.summarize(path == '/api/summary' && pair ? pair[1] : nil).to_json
          type = 'application/json; charset=utf-8'
        rescue ArgumentError
          status, body = 400, 'invalid-values'
        end
      end
      [status, { 'content-type' => type, 'content-length' => body.bytesize.to_s }, [body]]
    end
  end
end
